src/main/java/nl/thehyve/ocdu/controllers/TemplateController.java

```diff
@@ -0,0 +1,68 @@
+package nl.thehyve.ocdu.controllers;
+
+import nl.thehyve.ocdu.models.OCEntities.ClinicalData;
+import nl.thehyve.ocdu.models.OcUser;
+import nl.thehyve.ocdu.models.UploadSession;
+import nl.thehyve.ocdu.repositories.ClinicalDataRepository;
+import nl.thehyve.ocdu.services.DataService;
+import nl.thehyve.ocdu.services.OcUserService;
+import nl.thehyve.ocdu.services.OpenClinicaService;
+import nl.thehyve.ocdu.services.UploadSessionService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import javax.servlet.http.HttpSession;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Created by bo on 6/17/16.
+ */
+@Controller
+@RequestMapping("/template")
+public class TemplateController {
+
+    @Autowired
+    OcUserService ocUserService;
+
+    @Autowired
+    UploadSessionService uploadSessionService;
+
+    @Autowired
+    DataService dataService;
+
+    @Autowired
+    OpenClinicaService openClinicaService;
+
+    @Autowired
+    ClinicalDataRepository clinicalDataRepository;
+
+    @RequestMapping(value = "/check-new-patients", method = RequestMethod.GET)
+    public ResponseEntity<List<String>> checkNewPatients(HttpSession session) {
+        try {
+            UploadSession uploadSession = uploadSessionService.getCurrentUploadSession(session);
+            OcUser user = ocUserService.getCurrentOcUser(session);
+            String username = user.getUsername();
+            String pwdHash = ocUserService.getOcwsHash(session);
+            String url = user.getOcEnvironment();
+            List<ClinicalData> clinicalDatas = clinicalDataRepository.findBySubmission(uploadSession);
+
+            Map<String, String> subjectMap = openClinicaService.createMapSubjectLabelToSubjectOID(username, pwdHash, url, clinicalDatas);
```

Comment:
nope, the SOAP call createMapSubjectLabelToSubjectOID  still needs to be revisied with Jacob on Monday, I added a TODO comment as a reminder.